src/memory_manager.cc

```diff
@@ -101,7 +107,14 @@ MemoryManager::QueueMonitorThread()
 
       // Call the release callback.
       it->second->ReleaseCallback()(it->second->MemoryId());
+      it->second.reset();
       records_.erase(it);
+      {
+        bi::scoped_lock<bi::interprocess_mutex> lock{
+            *(ipc_message->ResponseMutex())};
+        memory_release_message_ptr->waiting_on_stub = true;
+        ipc_message->ResponseCondition()->notify_all();
```

Comment:
Why the previous approach didn't work?

Reply:
Synced offline. Revert the changes for extra round of message transfer. The queue of MemoryManager is of pointer type.